File: PL Final Project/student.c
#include <stdio.h>

struct student {
    char* name;
    int age; 
};
 
struct student student_student(char* n, int a) 
{
  
  struct student newStudent;
  newStudent.name = n; // probably want to copy all of the string
  newStudent.age = a;
  return newStudent; 

}

char* student_getName(struct student s) {
    return s.name; 
}

void student_setName(struct student *s, char* n) {
    //this.name = n;
    s->name = n;
}

int student_getAge(struct student s) {
    return s.age;
}

void student_SetAge(struct student *s, int a) {
    s->age = a; 
}


int main()
{
    struct student Jacob = student_student("jacob",100);
    struct student Wolfe = student_student("wolfe",999);
    struct student Luke = student_student("Luke",2);
    student_setAge (&Jacob,student_getAge(Wolfe) - student_getAge(Luke));
    
    printf("%d\n",student_getAge(Jacob)); 
    
    return 0;  

}

File: PL Final Project/student.java
class Student
{
    private String name;
    private int age;

    public Student (String n, int a)
    {
        name = n;
        age = a;
    }

    public String getName ()
    {
        return name;
    }

    public void setName (String n)
    {
        this.name = n;
    }

    public int getAge ()
    {
        return age;
    }

    public void setAge (int a)
    {
        this.age = a;
    }
}

public class main 
{
        public static void main(String[] args) 
    {
        Student Jacob = new Student("Jacob",100);
        Student Wolfe = new Student("Wolfe",999);
        Student Luke = new Student("Luke",2);

        Jacob.setAge(Wolfe.getAge() - Luke.getAge());

        System.out.print(Jacob.getAge()); 

    }
}

    © 2021 GitHub, Inc.
    Terms
    Privacy
